command/agent/retry_join_test.go

```diff
@@ -308,6 +313,60 @@ func TestRetryJoin_RetryMaxAttempts(t *testing.T) {
 	case <-timeout.C:
 		t.Fatal("timeout reached without error channel close")
 	}
+
+	must.True(t, strings.Contains(logOutput.String(), "join failed"), must.Sprint("expected log not found"))
```

Comment:
This and the other "expected log not found" assertion on the other test are failing.

Reply:
Yeah, I'm trying to determine why these tests are being so flaky as they are working as expected locally.

Reply:
To be honest, checking that the logging for this works as expected could probably be a bench test. Spending a lot of effort on this assertion might be worth the trouble.